### PR TITLE
chore: Setup for `CompletionRanker`

### PR DESCRIPTION
**Pull request by LukasRonn**
Initial skeleton for ranking completions.

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala [hunk 1]
```diff
@@ -15,16 +15,23 @@
...
+  */
+object DeltaContext {
+  /**
+    * Merges two DeltaContext.
+    *
+    * @param d1 the first DeltaContext.
+    * @param d2 the second DeltaContext.
+    * @return   a DeltaContext with a new Delta containing all changes
+    */
+  def mergeDeltas(d1: DeltaContext, d2: DeltaContext): DeltaContext = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala:34]: I guess this function is not commutative.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala [hunk 2]
```diff
@@ -15,16 +15,23 @@
  */
 package ca.uwaterloo.flix.api.lsp.provider.completion
 
-import ca.uwaterloo.flix.language.ast.Symbol
-
 /**
   * Represents a list of changes.
   */
-case class DeltaContext(deltas: List[Delta]) {
+case class DeltaContext(delta: Delta)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala:21]: Why have one class that wraps another?

Comment (LukasRonn) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala:21]: That is indeed not a clever solution.

File: main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala (hunk 1 above)
Comment (LukasRonn) [reply main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala:34]: Indeed not, I'll specify that.

**Pull request merged**